mono/metadata/threads.c

```diff
@@ -4999,6 +4999,11 @@ mono_threads_add_joinable_thread (gpointer tid)
 	 * 2fd16f60/r114307. So we collect them and join them when
 	 * we have time (in he finalizer thread).
 	 */
+	if (tid == NULL) {
+		g_warning ("Invalid thread: %p", tid);
```

Comment:
This should be a g_assert (tid);

Reply:
That would not provide a workaround for this bug: the build would continue to stop early (and would so, on even more platforms than it does so far) since `g_assert ()` definitely stops the program on any platform. https://github.com/luhenry/mono/blob/6928fda088c549648b0483e4e88c1d3dcdfdda30/mono/metadata/gc.c#L998 adds `0` to the list of threads which ends up here. The idea behind using `g_warning ()` was to have something that does not exit the runtime but still sends a strong message/warning.

After all, I really hope that https://github.com/mono/mono/pull/5549 succeeds somehow. This would make this PR irrelevant as it fixes the bug cleanly at its very root.